catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/DmsCoordinateProcessor.java

```diff
@@ -0,0 +1,198 @@
+      final Map<String, Object> dmsLat = new HashMap<String, Object>();
+      final Map<String, Object> dmsLon = new HashMap<String, Object>();
+      parseDms(dmsLat, dmsLon, dmsString);
+
+      final boolean dmsLatExists =
+          dmsLat.containsKey("degrees")
+              && dmsLat.containsKey("minutes")
+              && dmsLat.containsKey("seconds");
+      final boolean dmsLonExists =
+          dmsLon.containsKey("degrees")
+              && dmsLon.containsKey("minutes")
+              && dmsLon.containsKey("seconds");
+      return (dmsLatExists && dmsLonExists) ? toLatLon(dmsLat, dmsLon) : null;
+    }
+
+    private static LatLon toLatLon(
+        final Map<String, Object> dmsLat, final Map<String, Object> dmsLon) {
+      final int latModifier = dmsLat.get("direction").toString().toUpperCase().equals("N") ? 1 : -1;
+      final int lonModifier = dmsLon.get("direction").toString().toUpperCase().equals("E") ? 1 : -1;
+      final Double lat = toDecimalDegrees(dmsLat) * latModifier;
+      final Double lon = toDecimalDegrees(dmsLon) * lonModifier;
+
+      return (LatLon.isValidLatitude(lat) && LatLon.isValidLongitude(lon))
+          ? new LatLon(lat, lon)
+          : null;
+    }
+
+    private static Double toDecimalDegrees(final Map<String, Object> dms) {
+      final int degrees = (int) dms.get("degrees");
+      final int minutes = (int) dms.get("minutes");
+      final double seconds = (double) dms.get("seconds");
+
+      return degrees + minutes / 60.0 + seconds / 3600.0;
+    }
+  }
+
+  // This key tells the UI that the geo is on the suggestion itself
+  private static final String LITERAL_SUGGESTION_ID = "LITERAL-DMS";
+
+  /**
+   * Given a list of {@link Suggestion}s and the query that yielded them, enhance the list with
+   * additional suggestions based upon the presence of coordinate literals in the query string.
+   *
+   * @param results the list of results to enhance.
+   * @param query the user-provided gazetteer query text, which may contain coordinate literals.
+   */
+  public void enhanceResults(final List<Suggestion> results, final String query) {
+    LOGGER.trace("(DMS) Adding result for query [{}]", query);
+    final LiteralSuggestion dmsSuggestion = getDmsSuggestion(query);
+    if (dmsSuggestion != null && dmsSuggestion.hasGeo()) {
+      LOGGER.trace("Adding the DMS suggestion to results [{}]", dmsSuggestion);
+      results.add(0, dmsSuggestion);
+    }
+    LOGGER.trace("(DMS) Done");
+  }
+
+  @Nullable
+  private LiteralSuggestion getDmsSuggestion(final String query) {
+    final List<LatLon> dmsCoordinates = new ArrayList<>();
+    final Matcher matcher = PATTERN_DMS_COORDINATE.matcher(query);
+    final StringBuilder nameBuilder = new StringBuilder("DMS:");
+
+    int start = 0;
+    while (matcher.find()) {
+      LOGGER.trace("Match found [{}]", matcher.group());
+      final LatLon parsedDms = CoordinateTranslator.dmsToLatLon(query.substring(start, matcher.end()));
```

Comment:
I could do that, but then the processor would match a string like `140 3 3N 80 3 3E`  (it would suggest `40 3 3N...` , but degrees should be less than 90). Is there a maybe a way to change the regex?

Reply:
I'm confused. Why do we pass this substring to `dmsToLatLon()` but pass `matcher.group()` to `normalizedDmsString()`?

Reply:
It's confusing, I'll change it. But the reason why it works is that if the call to dmsToLatLon from the passed substring is not null, then passing matcher.group() into normalizedDmsStringparsedDms also will match the same thing. ParseDms calls matcher.matches(), while the loop above is calling matcher.find()